src/libasr/pass/print_list_constant.cpp

```diff
@@ -0,0 +1,132 @@
+      ASR::expr_t *comma = ASRUtils::EXPR(ASR::make_StringConstant_t(
+          al, x.base.base.loc, s2c(al, ", "), str_type3));
+      ASR::expr_t *single_quote = ASRUtils::EXPR(ASR::make_StringConstant_t(
+          al, x.base.base.loc, s2c(al, "'"), str_type2));
+      ASR::expr_t *empty_str = ASRUtils::EXPR(ASR::make_StringConstant_t(
+          al, x.base.base.loc, s2c(al, ""), str_type1));
+      ASR::expr_t *open_bracket = ASRUtils::EXPR(ASR::make_StringConstant_t(
+          al, x.base.base.loc, s2c(al, "["), str_type2));
+      ASR::expr_t *close_bracket = ASRUtils::EXPR(ASR::make_StringConstant_t(
+          al, x.base.base.loc, s2c(al, "]"), str_type2));
+      v1.push_back(al, comma);
+      v2.push_back(al, single_quote);
+      v3.push_back(al, open_bracket);
+      v4.push_back(al, close_bracket);
+
+      ASR::stmt_t *print_comma = LFortran::ASRUtils::STMT(ASR::make_Print_t(
+          al, x.base.base.loc, nullptr, v1.p, v1.size(), nullptr, empty_str));
+      ASR::stmt_t *print_open_bracket = LFortran::ASRUtils::STMT(
+          ASR::make_Print_t(al, x.base.base.loc, nullptr, v3.p, v3.size(),
+                            nullptr, empty_str));
+      ASR::stmt_t *print_close_bracket = LFortran::ASRUtils::STMT(
+          ASR::make_Print_t(al, x.base.base.loc, nullptr, v4.p, v4.size(),
+                            nullptr, empty_str));
+      ASR::stmt_t *print_single_quote = LFortran::ASRUtils::STMT(
+          ASR::make_Print_t(al, x.base.base.loc, nullptr, v2.p, v2.size(),
+                            nullptr, empty_str));
+      ASR::stmt_t *print_empty_endl =
+          LFortran::ASRUtils::STMT(ASR::make_Print_t(
+              al, x.base.base.loc, nullptr, nullptr, 0, nullptr, nullptr));
+
+      pass_result.push_back(al, print_open_bracket);
+      for (size_t j = 0; j < listC->n_args; ++j) {
+        if (j == 0) {
+          arg_type = expr_type(
+              listC->m_args[j]); // current the arg_type in List is same
+        } else {
+          pass_result.push_back(al, print_comma);
+        }
+        if (ASRUtils::is_character(*arg_type)) {
+          pass_result.push_back(al, print_single_quote);
+          pass_result.push_back(al,
+                                LFortran::ASRUtils::STMT(ASR::make_Print_t(
+                                    al, x.base.base.loc, nullptr,
+                                    &listC->m_args[j], 1, nullptr, empty_str)));
+          pass_result.push_back(al, print_single_quote);
+        } else if (ASRUtils::is_real(*arg_type) ||
+                   ASRUtils::is_integer(*arg_type)) {
+          pass_result.push_back(al,
+                                LFortran::ASRUtils::STMT(ASR::make_Print_t(
+                                    al, x.base.base.loc, nullptr,
+                                    &listC->m_args[j], 1, nullptr, empty_str)));
+        } else {
+          throw LCompilersException("Printing ConstantList is only support for"
```

Comment:
I am a bit confused here. Why the type of the list element is a constraint for printing? LLVM backend supports everything for printing and if not then also at least ASR shouldn’t be limited because of LLVM. Is there any blocker at ASR level that compelled you to add this check?

Reply:
Yes, In fact the day before I commit in on LLVM backend.
At that time I notice print statement not support with list_constant .
And the logic is almost the same as now, but in ASR pass now.
I just not know how to initilize a list_constant with complex item type and so on, and not include it in the test file the same as the implementation.

Reply:
What happens if you remove this check? Do the current tests pass? Which ones fail and what's the error message you receive?

Reply:
https://github.com/lcompilers/lpython/blob/d720d6abcf918106e2c9bd7a5f50bcd5db51c027/src/libasr/codegen/asr_to_llvm.cpp#L4559-L4693

The print ListConstant type is not supported in LLVM backend.
And I think you want to support an ASR pass is to transform the 
print list_constant ---to--->  print item ( n times)
which will fix other backend too, if we only support print listConstant in llvm backend.

Reply:
Now like the code, it will triger an error 

```
def f():
    print([complex(6, 8)])

f()
```

Reply:
I will need to add more on other type.

Reply:
Sure. Take your time.